src/ray/core_worker/reference_count.cc

```diff
@@ -915,11 +916,12 @@ bool ReferenceCounter::AddObjectLocation(const ObjectID &object_id,
   absl::MutexLock lock(&mutex_);
   auto it = object_id_refs_.find(object_id);
   if (it == object_id_refs_.end()) {
-    RAY_LOG(WARNING) << "Tried to add an object location for an object " << object_id
-                     << " that doesn't exist in the reference table";
+    RAY_LOG(INFO) << "Tried to add an object location for an object " << object_id
```

Comment:
Why is it info? Is it happening under normal condition?

Reply:
Yes, I noticed that this and the corresponding log statement in `RemoveObjectLocation` were getting logged in a few Python tests, where it looked like the owner was deleting the reference before an object location removal (and sometimes addition) was reported. My impression was that this ordering of events, at least for `RemoveObjectLocation`, popping up is expected given that the reference deletion at the end of `DeleteReferenceInternal` will almost always happen before 1. the `WaitForObjectEviction` RPC returns, 2. the object is freed from Plasma, 3. the deletion triggers a `RemoveObjectLocationsOwner` RPC to the owner, 4. this `RemoveObjectLocations` method is hit.

I'm pretty sure this could also happen for `AddObjectLocation`? Although I'm less sure about that.